tools/toolchains/__init__.py

```diff
@@ -849,6 +849,21 @@ def compile_command(self, source, object, includes):
 
         return None
 
+    def check_toolchain_path(self, name):
+        """Check if the path to toolchain is valid. Exit if not.
+
+        Positional arguments:
+        name -- the name of the toolchain, as used when creating compile
+                commands to execute. Valid key for TOOLCHAIN_PATHS.
+
+        No return value but causes a system exit of execution if the path
+        does not exist.
+        """
```

Comment:
@sarahmarshy It does exit gracefully with an error, but I think we typically try to stay away from calling `exit` within API functions (like toolchains or build_api). Usually I think we try to return error codes or throw exceptions in API functions and then let the top level CLI script that's calling them handle the exception and exit. Thoughts on this @theotherjimmy ?

@mazimkhan You're right I think this might cause issues with GCC_ARM because the toolchains can find it in the system PATH. I think this is only for GCC_ARM though unfortunately :/ @sarahmarshy sounds like another corner case.